src/main/java/duke/parser/CommandParser.java

```diff
@@ -0,0 +1,73 @@
+package duke.parser;
+
+import duke.entity.command.*;
+import duke.entity.task.Deadline;
+import duke.entity.task.Event;
+import duke.entity.task.Todo;
+import duke.handler.Ui;
+
+import java.text.ParseException;
+import java.util.Date;
+
+public class CommandParser {
+
+    private DateTimeParser dateTimeParser;
+    private Ui ui;
+
+    public CommandParser(Ui ui) {
+        this.dateTimeParser = new DateTimeParser();
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user input to determine which command to be created, together with required information to carry out task
+     *
+     * @param command next line of user input
+     * @return the Command object created based on user input
+     */
+    public Command parse(String command) {
```

Comment:
Instead of having this big chunk of code, you may want to break it into smaller functions. This is the application of Functional Abstraction. It makes your code more readable as well.

Reply:
Also, it is better to use enums for your command's keyword, instead of using if-else blocks, and trying to check whether the keyword is equal to some desired valid keywords.